crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -259,12 +449,12 @@ impl DerefMut for TimelineInnerState {
 }
 
 pub(in crate::timeline) struct TimelineInnerStateTransaction<'a> {
-    pub items: ManuallyDrop<ObservableVectorTransaction<'a, Arc<TimelineItem>>>,
+    pub items: ObservableVectorTransaction<'a, Arc<TimelineItem>>,
```

Comment:
Ah, I see. I think we need some safeguard against misuse of this, for the future:

- Does the `ObservableVectorTransaction` gets committed on drop, if it had changes that weren't committed, or unconditionally, or never?
- If unconditionally: is there an extra cost to having a commit with no new data updates?
- If never: can we have a small wrapper for this, that tracks whether it's been explicitly committed, using a boolean, and panics in Drop if it hasn't been explicitly committed?

Reply:
It rolls back in Drop. I don't think we should panic in our Drop, it could easily happen if we hold the txn over a `.await` and the future gets cancelled there. We could log I guess.

Reply:
Good point about cancellation. Warning sounds nice then 👍 (or debug_assert, for testing purposes?)